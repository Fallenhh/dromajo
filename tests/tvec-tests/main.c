//******************************************************************************
// Copyright (C) 2019, Esperanto Technologies Inc.
// The copyright to the computer program(s) herein is the
// property of Esperanto Technologies, Inc. All Rights Reserved.
// The program(s) may be used and/or copied only with
// the written permission of Esperanto Technologies and
// in accordance with the terms and conditions stipulated in the
// agreement/contract under which the program(s) have been supplied.
//------------------------------------------------------------------------------

/*

  Try writing some values to [sm]tvec and do an ecall.  This is for
  cosimulation checks.
*/

asm("        .section .text.startup,\"ax\",@progbits\n"
    "        .globl  _start\n"
    "        .type   _start, @function\n"
    "_start:\n"
    "        li	     a0, -1\n"
    "        csrw    mtvec, a0\n"
    "        csrr    a1, mtvec\n"
    "        csrw    stvec, a0\n"
    "        csrr    a1, stvec\n"

    "        li	     a0, -2\n"
    "        csrw    mtvec, a0\n"
    "        csrr    a1, mtvec\n"
    "        csrw    stvec, a0\n"
    "        csrr    a1, stvec\n"

    "        lui     a0,0x1feed\n"
    "        csrw    0x8d0,a0\n"

    "        .size   _start, .-_start\n");
