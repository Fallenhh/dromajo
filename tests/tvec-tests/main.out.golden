This emulator compiled with JTAG Remote Bitbang client. To enable, use +jtag_rbb_enable=1.
3 0x0000000000010000 (0xf1402573) x10 0x0000000000000000 csrr    a0, mhartid
3 0x0000000000010004 (0x00050663)                        beqz    a0, pc + 12
3 0x0000000000010010 (0x00000597) x11 0x0000000000010010 auipc   a1, 0x0
3 0x0000000000010014 (0x0f058593) x11 0x0000000000010100 addi    a1, a1, 240
3 0x0000000000010018 (0x60300413) x8  0x0000000000000603 li      s0, 1539
3 0x000000000001001c (0x7b041073)                        csrw    dcsr, s0
3 0x0000000000010020 (0x0010041b) x8  0x0000000000000001 addiw   s0, zero, 1
3 0x0000000000010024 (0x02741413) x8  0x0000008000000000 slli    s0, s0, 39
3 0x0000000000010028 (0x7b141073)                        csrw    dpc, s0
3 0x000000000001002c (0x7b200073)                        dret
3 0x0000008000000000 (0x0000557d) x10 0xffffffffffffffff c.li    a0, -1
3 0x0000008000000002 (0x30551073)                        csrw    mtvec, a0
3 0x0000008000000006 (0x305025f3) x11 0x000000ffffffff01 csrr    a1, mtvec
3 0x000000800000000a (0x10551073)                        csrw    stvec, a0
3 0x000000800000000e (0x105025f3) x11 0xffffffffffffff01 csrr    a1, stvec
3 0x0000008000000012 (0x00005579) x10 0xfffffffffffffffe c.li    a0, -2
3 0x0000008000000014 (0x30551073)                        csrw    mtvec, a0
3 0x0000008000000018 (0x305025f3) x11 0x000000fffffffffc csrr    a1, mtvec
3 0x000000800000001c (0x10551073)                        csrw    stvec, a0
3 0x0000008000000020 (0x105025f3) x11 0xfffffffffffffffc csrr    a1, stvec
3 0x0000008000000024 (0x1feed537) x10 0x000000001feed000 lui     a0, 0x1feed
ET validation mhartid=0 PASS code=0
3 0x0000008000000028 (0x8d051073)                        csrw    unknown_8d0, a0
Completed after 275 cycles
